converter/src/main/java/org/dswarm/converter/flow/TransformationFlow.java

```diff
@@ -351,30 +311,26 @@ public String apply(
 
 				TransformationFlow.LOG.error(message);
 
-				throw new DMPConverterException(message);
-			}
-		}
-
-		final String resultString;
-
-		try {
-
-			resultString = objectMapper.writeValueAsString(gdmModel.toJSON());
-		} catch (final JsonProcessingException e) {
+				resultObservable = Observable.error(new DMPConverterException(message));
 
-			final String message = "couldn't convert result into JSON";
+			}
 
-			TransformationFlow.LOG.error(message);
+		} else {
 
-			throw new DMPConverterException(message, e);
+			resultObservable = stateObservable;
 		}
 
-		gdmContext.stop();
+		tuples.subscribe(opener::process, writer::propagateError, opener::closeStream);
 
-		return resultString;
+		morphContext.stop();
+
+		return resultObservable
```

Comment:
I think that we need to change this blocking statement here, otherwise, we would also generate the full model even if we don't return it to the client.